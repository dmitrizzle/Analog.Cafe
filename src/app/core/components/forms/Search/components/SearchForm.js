import React from "react"

import { GA } from "../../../../../utils"
import Button from "../../../controls/Button/components/Button"
import Form from "../../../../../user/components/forms/Form"
import SearchInput from "../../SearchInput"

export default class extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      query: "",
      warning: false
    }
  }
  handleEmailChange = event => {
    this.setState({ query: event.target.value || "", warning: false })
    this.props.searchText(event.target.value)
  }
  handleSubmit = event => {
    event.stopPropagation()
    event.preventDefault()
    this.props.submitCallback && this.props.submitCallback(this.state.query)
    GA.event({
      category: "Navigation",
      action: "SearchForm_send",
      label: this.props.formLocation ? this.props.formLocation : null
    })
  }
  handleInputClick = event => {
    event.stopPropagation()
  }
  render = () => {
    return (
      <Form
        style={this.props.style || null}
        withinGroup={this.props.withinGroup}
      >
        <SearchInput
          onChange={this.handleEmailChange}
          warning={this.state.warning}
          autoFocus={this.props.autoFocus}
          onClick={this.handleInputClick}
        />
        <Button
          inverse
          onClick={event => {
            this.handleSubmit(event)
          }}
          loading={this.props.loading}
        >
          {this.props.buttonText || "Search"}
        </Button>
      </Form>
    )
  }
}
