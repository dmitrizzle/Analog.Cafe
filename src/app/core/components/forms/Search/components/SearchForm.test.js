import "jest-styled-components"

import { shallow } from "enzyme"
import React from "react"

import SearchForm from "./SearchForm"

it("Render SearchForm, matches snapshot", () => {
  const element = shallow(<SearchForm />)
  expect(element).toMatchSnapshot()
})
