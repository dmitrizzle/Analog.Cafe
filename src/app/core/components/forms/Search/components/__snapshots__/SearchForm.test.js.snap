// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SearchForm, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          autoFocus={undefined}
          onChange={[Function]}
          onClick={[Function]}
          warning={false}
/>,
        <Unknown
          inverse={true}
          loading={undefined}
          onClick={[Function]}
>
          Search
</Unknown>,
      ],
      "style": null,
      "withinGroup": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "autoFocus": undefined,
          "onChange": [Function],
          "onClick": [Function],
          "warning": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Search",
          "inverse": true,
          "loading": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Search",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            autoFocus={undefined}
            onChange={[Function]}
            onClick={[Function]}
            warning={false}
/>,
          <Unknown
            inverse={true}
            loading={undefined}
            onClick={[Function]}
>
            Search
</Unknown>,
        ],
        "style": null,
        "withinGroup": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "autoFocus": undefined,
            "onChange": [Function],
            "onClick": [Function],
            "warning": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Search",
            "inverse": true,
            "loading": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
